### PR TITLE
feat: prometheus endpoint

### PR DESCRIPTION
**Pull request by monoid**

### CODE REVIEW

File: ccp/src/hashrate/prometheus.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+) -> response::Result<http::Response<body::Body>> {
+    let mut buf = String::new();
+
+    {
+        let mut registry = Registry::default();
+
+        {
+            let guard = state.hashrate_collector.lock().map_err(|_| {
+                // TODO use parking_lot mutex
+                log::warn!("Prometehus metrics lock is poisoned");
```
Comment (monoid) [inline ccp/src/hashrate/prometheus.rs:53]: Should be `error!`, as it is fatal.

File: crates/config/src/unresolved_config.rs [hunk 2]
```diff
@@ -31,6 +31,7 @@ const DEFAULT_UTILITY_THREAD_ID: u32 = 1;
 #[serde(rename_all = "kebab-case")]
 pub struct UnresolvedCCPConfig {
     pub http_server: UnresolvedHTTPServer,
+    pub prometheus_endpoint: UnresolvedPrometheusEndpoint,
```
Comment (mikevoronov) [inline crates/config/src/unresolved_config.rs:34]: I suggest making prometheus metrics optional, so we can run CCP with disabled PrometheusClient.

Comment (monoid) [reply crates/config/src/unresolved_config.rs:34]: Makes sense.

**Review by mikevoronov — APPROVED** [review]
utACK

**Pull request merged**